tests/libvmi/factory.go

```diff
@@ -47,27 +46,49 @@ func NewFedora(opts ...Option) *kvirtv1.VirtualMachineInstance {
 
 // NewCirros instantiates a new CirrOS based VMI configuration
 func NewCirros(opts ...Option) *kvirtv1.VirtualMachineInstance {
+	diskImage := WithContainerImage(cd.ContainerDiskFor(cd.ContainerDiskCirros))
+
+	opts = append(opts, diskImage)
+	cirrosOpts := NewCirrosOptions(opts...)
+
+	return New(RandName(DefaultVmiName), cirrosOpts...)
+}
+
+// NewCirrosOptions instantiates default options needed for a CirrOS based VMI configuration
+func NewCirrosOptions(opts ...Option) []Option {
```

Comment:
I find it unusual, that this function takes a slice parameter and only adds it to the resulting slice. It could just return the `cirrosOpts` and the calling function would add them to other parameters.
Is it important that the cloudInit disk is last?

Reply:
The main reason for such a function is that cloudInit had to be the last disk. So it takes all the parameters and applies everything in order.

I'll rebase and test it again to make sure this is the reason it failed. If cloudInit does not need to be last then I'll change the code so it simply returns options.

Reply:
We could use `bootOrder` if we for whatever reason don't want to rely on the order(of options)...

Reply:
ok, confirmed, the hacky cloudInit has to be last.
So, I would need to add a bootOrder to the disk? I need to check how to do this correctly so other disks, any disks added by end user are not in conflict.

Reply:
I am not sure it will be easier or cleaner, I would need to provide a consistent way to order all boot devices when creating cirros, because we want the cloudInit to be the last. Of course it will be more **correct** than ordering it in yaml. 

Anyway, I am experimenting with it right now.

Reply:
From my experiments it looks like Boot order only works if applied to all devices. So to correctly create Cirros I would have to get the vmi spec, then order all disks, and then make sure the cloudInit is last. I can add this.

Reply:
I have actually changed it so it returns options. And I have also changed the way we add boot disk. Now the last one wins (replaces previous one) so we can have defaults and it can be easily replaced by user provided options.